core/store/src/trie/split_state.rs

```diff
@@ -67,68 +146,125 @@ impl ShardTries {
         Ok((store_update, new_state_roots))
     }
 
+    fn get_trie_updates(
+        &self,
+        state_roots: &HashMap<ShardUId, StateRoot>,
+    ) -> HashMap<ShardUId, TrieUpdate> {
+        state_roots
+            .iter()
+            .map(|(shard_uid, state_root)| {
+                (*shard_uid, self.new_trie_update(*shard_uid, state_root.clone()))
+            })
+            .collect()
+    }
+
     pub fn apply_delayed_receipts_to_split_states<'a>(
         &self,
         state_roots: &HashMap<ShardUId, StateRoot>,
-        receipts: &[Receipt],
+        receipts: Vec<Receipt>,
```

Comment:
Hm, why do we need a Vec here, seems [] is enough?

Reply:
Yeah, I just saw that everywhere this function is called, `Vec<Receipt>` is used, so I changed it. I just changed it back